Lib/unittest/case.py

```diff
@@ -603,9 +603,23 @@ def _callSetUp(self):
         self.setUp()
 
     def _callTestMethod(self, method):
-        if method() is not None:
-            warnings.warn(f'It is deprecated to return a value that is not None from a '
-                          f'test case ({method})', DeprecationWarning, stacklevel=3)
+        if (result := method()) is not None:
+            import inspect
+            from unittest.async_case import IsolatedAsyncioTestCase
+            msg = (
+                'It is deprecated to return a value that is not None, '
+                f'got: ({type(result)}) from a '
```

Comment:
I think we usually just quote the name of the class in user-facing messages, e.g.

```
% py -c '"x" + 42'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
TypeError: can only concatenate str (not "int") to str
```

```suggestion
                f'got: ({type(result).__name__!r}) from a '
```